webpack/assets/javascripts/react_app/redux/middlewares/DebounceMiddleware/DebounceMiddleware.js

```diff
@@ -0,0 +1,34 @@
+import { omit } from 'lodash';
+import { selectDebounceItem } from './DebounceSelectors';
+import {
+  startDebounce,
+  clearDebounce,
+  stopIncomingAction,
+} from './DebounceActions';
+
+export const DebounceMiddleware = store => next => action => {
+  const { key, debounce, payload = {} } = action;
+  const debounceKey = key || payload.key;
```

Comment:
Why do we need to support both `key` and `payload.key`? Can't we use something more declarative like `action.debounceKey`?

Reply:
since there can be a combination of multiple middlewares where all are using the `key` key,
e.g: passing into the interval middleware an API action with debounce

this line is here to make both action structures to work: `{ type, key, payload }` || `{ type, payload: { key }}`